### PR TITLE
Add update page for expiring credit cards

### PR DESCRIPTION
**Pull request by sbinlondon**
Relates opencollective/opencollective#2683 

There's lots of cleaning up (logs, variables, etc) to be done on the PR but wanted to get it in front of you for feedback first and most importantly! 🙈

* The `UpdatePaymentPage` will link from the email and include the user slug and payment method id in the URL
* Using this information we can load the user's active subscriptions to tell them what will be updated
* They then fill out the form with the new credit card information, and this is fed to the API which then updates all subscriptions attached to the old payment method using the new 

Things I know I need to fix:

* Load only subscriptions connected to the card (right now it displays all active subs)
* Better error handling - there is no error view now if the backend work goes wrong
* Better UX after the card details have been updated (i.e. remove the disabled button and put a message)

Please do add your questions and feedback for making this better and in a good state to be merged later.

Dec-06-2019 09-01-27

(Will also clean up commits later of course - just left a trail in case I needed to backtrack!)

### CODE REVIEW

**Comment by znarf** [comment]
There is some interesting feedback from the linter. You don't have that integrated in your editor?

Otherwise:
- `npm run langs:update` to make CI happy
- `npm run graphql:update` to make the GraphQL linter happy

File: pages/updatePaymentMethod.js [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+
+const ShadowBox = styled(Box)`
+  box-shadow: 0px 8px 16px rgba(20, 20, 20, 0.12);
+`;
+
+const Subtitle = styled(H5)`
+  color: white;
+  text-align: center;
+  margin: 0 auto;
+  ${fontSize};
```
Comment (Betree) [inline pages/updatePaymentMethod.js]: This prop is already supported natively by `H5`

File: pages/updatePaymentMethod.js [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+    /** From intl */
+    intl: PropTypes.object.isRequired,
+    /** From graphql query */
+    replaceCreditCard: PropTypes.func.isRequired,
+    /** From stripeLoader */
+    loadStripe: PropTypes.func.isRequired,
+  };
+
+  state = {
+    loading: true,
```
Comment (Betree) [inline pages/updatePaymentMethod.js]: I don't think this is used

File: pages/updatePaymentMethod.js (hunk 1 above)
Comment (sbinlondon) [reply pages/updatePaymentMethod.js]: What do you mean? To not use the fontSize and maxWidth from the styled-system? I have tried to add these props to the Subtitle directly and they don't seem to be applying 🤔

File: server/pages.js [hunk 3]
```diff
@@ -11,6 +11,7 @@ const pages = routes()
   .add('static', '/:pageSlug(widgets|tos|privacypolicy|support|hiring)', 'staticPage')
   .add('redeem', '/:collectiveSlug?/redeem/:code?')
   .add('redeemed', '/:collectiveSlug?/redeemed/:code?')
+  .add('updatePaymentMethod', '/paymentmethod/:id/:collectiveSlug/update')
```
Comment (znarf) [inline server/pages.js]: If we put the `collectiveSlug` in the URL then it should be at the beginning.

File: pages/updatePaymentMethod.js [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+      },
+    };
+  },
+
+  props: ({ data }) => {
+    let allSubs = [];
+    let subscriptions = [];
+
+    if (data && data.Collective) {
+      allSubs = cloneDeep(data.Collective.ordersFromCollective);
```
Comment (znarf) [inline pages/updatePaymentMethod.js]: `cloneDeep` should not be necessary here and the `allSubs` is then not needed.

File: components/EditPaymentMethod.js [hunk 5]
```diff
@@ -90,6 +90,8 @@ class EditPaymentMethod extends React.Component {
     const saved = this.state.paymentMethod.monthlyLimitPerMember === paymentMethod.monthlyLimitPerMember;
     const hasActions = !saved || canRemove || hasSubscriptions;
 
+    console.log(paymentMethod);
+
```
Comment (znarf) [inline components/EditPaymentMethod.js]: Should be removed.

File: pages/updatePaymentMethod.js (hunk 1 above)
Comment (znarf) [reply pages/updatePaymentMethod.js]: I think @Betree is just saying that ` ${fontSize};` can be removed because it's already supported in `H5`.

**Comment by sbinlondon** [comment]
Thanks for the comments, will update tomorrow 😊

**Comment by znarf** [comment]
Marking that as stale just for administrative purpose, this should be continued soon!

**Comment by sbinlondon** [comment]
Notes from today:

* center the header
* why is it missing the subscription information?
* update button should be further down in the white part
* HTML list about each sub: name of the collective, value of subscription, frequency
* address should be this order: localhost:3000/(user)/paymentmethod/50/update
* if user isn't authenticated, then we need to show the sign in form
* loading on client side until we know if authenticated: then load sign in or payment udpate page
* follow best practices on pages/editCollective.js

File: pages/updatePaymentMethod.js [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+                <Flex justifyContent="center" alignItems="center" flexDirection="column">
+                  <Container background="white" borderRadius="16px" maxWidth="600px">
+                    <ShadowBox py="24px" px="32px" minWidth="500px">
+                      {showCreditCardForm && (
+                        <Box mr={2} css={{ flexGrow: 1 }}>
+                          <NewCreditCardForm
+                            name="newCreditCardInfo"
+                            profileType={get(this.props.collective, 'type')}
+                            hasSaveCheckBox={false}
+                            // error={errors.newCreditCardInfo}
```
Comment (Betree) [inline pages/updatePaymentMethod.js]: Leftover?

Comment (sbinlondon) [reply pages/updatePaymentMethod.js]: Yes! I will remove it.

**Pull request merged**